Python 3.4.2 fails to install because of missing openssl lib

I do have a newer version of openssl installed from brew-dupes so that security is better, but that shouldn't ruin it.
Here's my puppet stuff: 

```puppet
class people::dkowis::pythons {
  python::version { '2.7.8': }
  python::version { '3.4.2': }

  class { 'python::global':
    version => '2.7.8',
  }
}
```
I get this result from running boxen:
```
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Downloading readline-6.3.tar.gz...
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: -> http://ftpmirror.gnu.org/readline/readline-6.3.tar.gz
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Installing readline-6.3...
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Installed readline-6.3 to /opt/boxen/pyenv/versions/3.4.2
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Downloading Python-3.4.2.tgz...
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: -> https://yyuu.github.io/pythons/44a3c1ef1c7ca3e4fd25242af80ed72da941203cb4ed1a8c1b724d9078965dd8
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Installing Python-3.4.2...
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Please consult to the Wiki page to fix the problem.
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: https://github.com/yyuu/pyenv/wiki/Common-build-problems
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: BUILD FAILED (OS X 10.10.3 using python-build 20141118)
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Inspect or clean up the working tree at /tmp/python-build.20150701140946.71921
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Results logged to /tmp/python-build.20150701140946.71921.log
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Last 10 log lines:
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: (cd /opt/boxen/pyenv/versions/3.4.2/share/man/man1; ln -s python3.4.1 python3.1)
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: if test "xupgrade" != "xno"  ; then \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 		case upgrade in \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 			upgrade) ensurepip="--upgrade" ;; \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 			install|*) ensurepip="" ;; \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 		esac; \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 		 ./python.exe -E -m ensurepip \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 			$ensurepip --root=/ ; \
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: 	fi
Notice: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: Ignoring ensurepip failure: pip 1.5.6 requires SSL/TLS
Error: /opt/boxen/pyenv/bin/pyenv install --skip-existing 3.4.2 returned 1 instead of one of [0]
Error: /Stage[main]/People::Dkowis::Pythons/Python::Version[3.4.2]/Exec[python-install-3.4.2]/returns: change from notrun to 0 failed: /opt/boxen/pyenv/bin/pyenv install --skip-existing 3.4.2 returned 1 instead of one of [0]
```

--- a/data/common.yaml
+++ b/data/common.yaml
@@ -10,6 +10,6 @@ python::version::alias:
 python::prefix: '/opt'
 python::user: "%{::id}"
 
-python::pyenv::ensure: v20150913
+python::pyenv::ensure: v20151103
 python::pyenv::prefix: "%{hiera('python::prefix')}/pyenv"
 python::pyenv::user: "%{hiera('python::user')}"

